File: RLang/ADMM_CNJGL/soft_thresh.R
soft_thresh <- function(Y, lambda) {
  b <- abs(Y) - lambda
  for (c in 1:dim(b)[2]) {
    for (r in 1:dim(b)[1]) {
      if (b[r,c] < 0) {b[r,c] = 0}                  
    }
  }
  return(sign(Y) * b)
}

File: PNJGL_onweb/ADMM_PNJGL/soft_thresh.m
%% -------------------- SOFT-THRESHOLDING OPERATOR ------------------- %%


% ---------------- minimize 1/2*||X - Y||_2^2 + lambda*||X||_1 -------- %



% ----------------- LAST UPDATE: 4/19/2012 ---------------------------- %



function [X] = soft_thresh(Y,lambda)

if(nargin == 1)
    disp('Not enough inputs');
    disp('Enter both Y and lambda');
    return;
end;
        
X = sign(Y).*(max(abs(Y) - lambda,0));
